src/programs/city/mine.js

```diff
@@ -100,7 +101,34 @@ class CityMine extends kernel.process {
       return
     }
 
-    const storage = this.room.storage
+    let storage = false
+    if (this.room.storage) {
+      storage = this.room.storage
+    } else if (this.room.terminal) {
+      storage = this.room.terminal
+    } else if (this.remote) {
+      const containers = this.room.structure[STRUCTURE_CONTAINER]
+      if (containers && containers.length > 0) {
+        if (containers.length > 1) {
+          containers.sort((a, b) => a.store[RESOURCE_ENERGY] - b.store[RESOURCE_ENERGY])
+        }
+        storage = containers[0]
+      } else {
```

Comment:
I think there was meant to be an extra closing curly bracket here before this else block, so it would continue the chain from line 109 instead of 111.

Reply:
No, this is right. The block that begins at 109 is the "we are mining a remote room but don't have any storage or terminal". It then picks a container to dump energy in, and if there are no containers it'll dump it in a spawn.

The reason this is only for remote rooms is that in local rooms there will already be containers (from the miners) and spawning haulers just to maybe move things to the spawn (which fillers will already do) has no value.